src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -373,9 +385,15 @@ protected String getString(BsonValue o) throws SQLException {
         if (checkNull(o)) {
             return null;
         }
+        JsonWriterSettings settings = JsonWriterSettings.builder().outputMode(JsonMode.EXTENDED).build();
+        CodecRegistry registry = fromProviders(new BsonValueCodecProvider());
         switch (o.getBsonType()) {
```

Comment:
I am wondering if it would not make sense to convert all values using extended Json in relaxed mode no matter the type. This would make things more coherent and predictable for everyone.

Reply:
Discussed offline and will decide how to handle this as part of the type mapping.